checked/ref.go

```diff
@@ -169,3 +171,37 @@ func (c *RefCount) DecWrites() {
 func (c *RefCount) NumWriters() int {
 	return int(atomic.LoadInt32(&c.writes))
 }
+
+// TrackObject sets up the initial internal state of the Ref for
+// leak detection.
+func (c *RefCount) TrackObject(v interface{}) {
+	if atomic.LoadUint64(&leakDetectionFlag) == 0 {
+		return
+	}
```

Comment:
This proposed API seems backwards – the natural impulse for anyone reading such code would be to move the `ShouldTrackObject()` check _into_ the `TrackObject()` since there's no other use for that function, also it's a little bit too late to think about performance, is it not? Of course, I also hope that this whole debugging thing is a temporary intstrumentation so making a few unexpected code design choices scoped to it is fine by me if you insist.